filament/backend/src/opengl/OpenGLDriver.cpp

```diff
@@ -3229,11 +3211,9 @@ void OpenGLDriver::draw(PipelineState state, Handle<HwRenderPrimitive> rph) {
 
     // If necessary, mutate the bindings in the VAO.
     VertexBufferHandle vbwo = rp->gl.vertexBufferWithObjects;
-    if (UTILS_UNLIKELY(vbwo)) {
-        const GLVertexBuffer* vb = handle_cast<GLVertexBuffer*>(vbwo);
-        if (rp->gl.vertexBufferVersion != vb->bufferObjectsVersion) {
-            updateVertexArrayObject(rp, vb);
-        }
+    const GLVertexBuffer* vb = handle_cast<GLVertexBuffer*>(vbwo);
+    if (UTILS_UNLIKELY(rp->gl.vertexBufferVersion != vb->bufferObjectsVersion)) {
```

Comment:
we're sure that `vb` can't be `nullptr` here? by construction?

Reply:
That's a good point, it's possible for Filament to create a render primitive and never set it up with a vertex buffer and index buffer.  However I would argue that calling draw() with an uninitialized render primitive is a mis-use of the Driver API.

Maybe this is a good candidate for assert_invariant.

Reply:
No I think it make more sense to do nothing in that case. It's like drawing an empty draw call.

Reply:
Ok, added an 'if unlikely" for the empty draw call case.